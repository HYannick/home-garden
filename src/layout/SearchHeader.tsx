/** @jsx jsx */
import React from 'react';
import { css, jsx } from '@emotion/core';
import styled from '@emotion/styled';
import { Formik } from 'formik';
import { useTranslation } from 'react-i18next';
import { Overlay } from '../components/Overlay';
import ImageFade from '../components/image-fade/ImageFade';
import SearchIcon from '../core/svg/Search';
import ActionBar from './ActionBar';

interface HeaderProps {
  onSubmit: (value: string) => void,
  cover?: string
}

export const SearchInput = styled('div')`
  position: absolute;
  bottom: -2.5rem;
  width: 100%;
  left: 0;
  right: 0;
  padding: 0 2rem;
  label {
    color: ${({ theme }) => theme.palette.light};
    font-size: 2rem;
    text-align: center;
    width: 100%;
    display: block;
    margin-bottom: 1.5rem;
  }
  input {
    width: 100%;
    padding: 2rem 6rem 2rem 2rem;
    border-radius: 3rem;
    border: none;
    font-size: 1.8rem;
    box-shadow: 0 0 1.5rem 0 #a7a7a754;
    outline: ${({ theme }) => theme.palette.grey.light};
    &::-webkit-search-cancel-button {
      display: none;
    }
  }
`;

const ActionbarWrapper = styled('div')`
  position: absolute;
  top: 0;
  z-index: 10;
  svg path {
    fill: ${({ theme }) => theme.palette.light};
    stroke: ${({ theme }) => theme.palette.light};
  }
`;

const SearchControl = styled('div')`
  position: relative;
`;
const IconButton = styled('button')`
  position: absolute;
  right: 1rem;
  top: 50%;
  border: none;
  outline: ${({ theme }) => theme.palette.grey.dark};
  transform: translateY(-50%);
  width: 5rem;
  height: 5rem;
  border-radius: 3rem;
  background-color: transparent;
  svg {
    width: 2.5rem;
    height: 2.5rem;
    path {
      fill: none;
      stroke-width: 0.3rem;
      stroke: ${({ theme }) => theme.palette.grey.dark};
    }
  }
`;

const Form = styled('form')`
  position: relative;
`;

const SearchHeader: React.FC<HeaderProps> = ({ onSubmit, cover }) => {
  const { t } = useTranslation();
  return (
    <div css={css`
      position: relative;
      height: 12rem;
      display: flex;
      align-items: center;
      margin-bottom: 5rem;
    `}>
      <div css={css`
        position: absolute;
        overflow: hidden;
        border-radius: 0 0 4.5rem 4.5rem;
        top: 0;
        left:0;
        bottom:0;
        right:0;
        z-index: -1;
      `}>
        <Overlay color="#333" opacity={0.7}/>
        <ImageFade src={cover} alt="cover" placeholder="#EFFFE2"/>
      </div>
      <ActionbarWrapper>
        <ActionBar/>
      </ActionbarWrapper>
      <SearchInput>
        <Formik
          initialValues={{ query: '' }}
          onSubmit={({ query }, actions) => {
            onSubmit(query);
            actions.setSubmitting(false);
          }}
          render={props => (
            <Form onSubmit={props.handleSubmit}>
              <label htmlFor="search">{t('search.title')}</label>
              <SearchControl>
                <input
                  type="search"
                  autoComplete="off"
                  id="search"
                  placeholder={t('search.placeholder')}
                  onChange={props.handleChange}
                  onBlur={props.handleBlur}
                  value={props.values.query}
                  name="query"
                />
                {props.errors.query && <div id="feedback">{props.errors.query}</div>}
                <IconButton type="submit"><SearchIcon/></IconButton>
              </SearchControl>
            </Form>
          )}
        />
      </SearchInput>
    </div>
  );
};
export default SearchHeader;
