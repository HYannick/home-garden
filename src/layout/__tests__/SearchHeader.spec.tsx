import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderWithTheme } from '../../theme-wrapper';
import SearchHeader from '../SearchHeader';

afterEach(cleanup);
describe('SearchHeader', () => {
  const props = {
    onSubmit: jest.fn(),
    cover: 'kitten.jpg',
  };
  it('should render properly', () => {
    const { container } = renderWithTheme(
      <Router><SearchHeader {...props}/></Router>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render the right label', () => {
    const { getByText } = renderWithTheme(
      <Router><SearchHeader {...props}/></Router>,
    );
    expect(getByText(/^search.title/)).toHaveTextContent('search.title');
  });
});
