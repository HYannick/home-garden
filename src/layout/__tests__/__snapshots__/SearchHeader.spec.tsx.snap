// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader should render properly 1`] = `
<div>
  <div
    class="css-ob4x19-SearchHeader"
  >
    <div
      class="css-5tnmxt-SearchHeader"
    >
      <div
        class="css-ilwsms"
        color="#333"
        opacity="0.7"
      />
      <div
        class="css-prjb17"
        placeholder="#EFFFE2"
      >
        <img
          alt="cover"
          class="css-10dr8mc"
          src="kitten.jpg"
        />
      </div>
    </div>
    <div
      class="css-1fzrr71"
    >
      <div
        class="css-11xigr1"
      >
        <div
          class="css-10wldv8"
        >
          <svg
            viewBox="0 0 23.225 21.762"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.342 19.647L2.115 10.881m0 0l10.227-8.766m-8.617 9.032h18"
              fill="#4A4A4A"
              stroke="#4A4A4A"
              stroke-linecap="round"
              stroke-width="3"
            />
          </svg>
        </div>
        <div
          class="css-196st0l"
        >
          <span
            class="css-zrqali"
          />
        </div>
        <div
          class="css-ov1ktg"
        />
      </div>
    </div>
    <div
      class="css-1of5vee"
    >
      <form
        class="css-cssveg"
      >
        <label
          for="search"
        >
          search.title
        </label>
        <div
          class="css-cssveg"
        >
          <input
            autocomplete="off"
            id="search"
            name="query"
            placeholder="search.placeholder"
            type="search"
            value=""
          />
          <button
            class="css-36uvdm"
            type="submit"
          >
            <svg
              viewBox="0 0 23.837 23.837"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 14l8.335 8.335zM1.5 8.792a7.293 7.293 0 1 1 7.292 7.293A7.292 7.292 0 0 1 1.5 8.792z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
